### PR TITLE
Exclude operators that produce unbacked symbols

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #120917

Unbacked symbols vary at runtime which means they are not CUDA
graphable.

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire

### CODE REVIEW

File: torch/_inductor/utils.py [hunk 1]
```diff
@@ -667,6 +669,8 @@ def has_incompatible_cudagraph_ops(gm):
     for node in gm.graph.nodes:
         if str(node.target) in forbidden_set:
             return True
+        if (val := node.meta.get("val")) is not None and free_unbacked_symbols(val):
```
Comment (eellison) [inline torch/_inductor/utils.py:672]: Can you log stack trace here to `disable_cudagraphs_reason` ?

**Review by eellison — APPROVED** [review]
actually just land, I need to rebase and land https://github.com/pytorch/pytorch/pull/118113/ for stack trace to work here

**Review by eellison — APPROVED** [review]
actually just land, I need to rebase and land https://github.com/pytorch/pytorch/pull/118113/ for stack trace to work here

**Review by eellison — COMMENTED** [review]
I think it would still be better to disable this elsewhere, because it's possible we fuse this unbacked symint entirely within a kernel, in which case it would work with cudagraphs

**Comment by ezyang** [comment]
> I think it would still be better to disable this elsewhere, because it's possible we fuse this unbacked symint entirely within a kernel, in which case it would work with cudagraphs

Yeah, so for that, it would have to be some sort of tag on the op, there's no way to know otherwise. Probably a completely new tag too...

**Pull request closed without merging**